docs/proposed/config-split.md

```diff
@@ -0,0 +1,420 @@
+//#define MBEDTLS_MEMORY_DEBUG
+//#define MBEDTLS_MEMORY_BACKTRACE
+//#define MBEDTLS_THREADING_ALT
+//#define MBEDTLS_THREADING_PTHREAD
+#define MBEDTLS_PLATFORM_C
+//#define MBEDTLS_THREADING_C
+#define MBEDTLS_TIMING_C
+//#define MBEDTLS_TIMING_ALT
+//#define MBEDTLS_PLATFORM_STD_MEM_HDR   <stdlib.h>
+//#define MBEDTLS_PLATFORM_STD_CALLOC        calloc
+//#define MBEDTLS_PLATFORM_STD_FREE            free
+//#define MBEDTLS_PLATFORM_STD_SETBUF      setbuf
+//#define MBEDTLS_PLATFORM_STD_EXIT            exit
+//#define MBEDTLS_PLATFORM_STD_TIME            time
+//#define MBEDTLS_PLATFORM_STD_FPRINTF      fprintf
+//#define MBEDTLS_PLATFORM_STD_PRINTF        printf
+//#define MBEDTLS_PLATFORM_STD_SNPRINTF    snprintf
+//#define MBEDTLS_PLATFORM_STD_EXIT_SUCCESS       0
+//#define MBEDTLS_PLATFORM_STD_EXIT_FAILURE       1
+//#define MBEDTLS_PLATFORM_STD_NV_SEED_READ   mbedtls_platform_std_nv_seed_read
+//#define MBEDTLS_PLATFORM_STD_NV_SEED_WRITE  mbedtls_platform_std_nv_seed_write
+//#define MBEDTLS_PLATFORM_STD_NV_SEED_FILE  "seedfile"
+//#define MBEDTLS_PLATFORM_CALLOC_MACRO        calloc
+//#define MBEDTLS_PLATFORM_FREE_MACRO            free
+//#define MBEDTLS_PLATFORM_EXIT_MACRO            exit
+//#define MBEDTLS_PLATFORM_SETBUF_MACRO      setbuf
+//#define MBEDTLS_PLATFORM_TIME_MACRO            time
+//#define MBEDTLS_PLATFORM_TIME_TYPE_MACRO       time_t
+//#define MBEDTLS_PLATFORM_FPRINTF_MACRO      fprintf
+//#define MBEDTLS_PLATFORM_PRINTF_MACRO        printf
+//#define MBEDTLS_PLATFORM_SNPRINTF_MACRO    snprintf
+//#define MBEDTLS_PLATFORM_VSNPRINTF_MACRO    vsnprintf
+//#define MBEDTLS_PLATFORM_NV_SEED_READ_MACRO   mbedtls_platform_std_nv_seed_read
+//#define MBEDTLS_PLATFORM_NV_SEED_WRITE_MACRO  mbedtls_platform_std_nv_seed_write
+//#define MBEDTLS_PLATFORM_MS_TIME_TYPE_MACRO   int64_t
+//#define MBEDTLS_PRINTF_MS_TIME    PRId64
+//#define MBEDTLS_MEMORY_ALIGN_MULTIPLE      4
+
+* SECTION "General and test configuration options"
+//#define MBEDTLS_PSA_CRYPTO_CONFIG_FILE "psa/crypto_config.h"
+//#define MBEDTLS_PSA_CRYPTO_USER_CONFIG_FILE "/dev/null"
+//#define MBEDTLS_DEPRECATED_WARNING
+//#define MBEDTLS_DEPRECATED_REMOVED
+//#define MBEDTLS_CHECK_RETURN_WARNING
+#define MBEDTLS_SELF_TEST
+//#define MBEDTLS_TEST_CONSTANT_FLOW_MEMSAN
+//#define MBEDTLS_TEST_CONSTANT_FLOW_VALGRIND
+//#define MBEDTLS_TEST_HOOKS
+
+//#define MBEDTLS_CHECK_RETURN __attribute__((__warn_unused_result__))
+//#define MBEDTLS_IGNORE_RETURN( result ) ((void) !(result))
+
+* SECTION "Cryptographic mechanism selection (PSA API)"
+PSA_WANT_\* macros as in current crypto_config.h.
+
+
+* SECTION "Cryptographic mechanism selection (extended API)"
+#define MBEDTLS_CIPHER_C
+//#define MBEDTLS_CTR_DRBG_USE_128_BIT_KEY
+#define MBEDTLS_CTR_DRBG_C
+#define MBEDTLS_HMAC_DRBG_C
+#define MBEDTLS_LMS_C
+//#define MBEDTLS_LMS_PRIVATE
+#define MBEDTLS_MD_C
+#define MBEDTLS_NIST_KW_C
+#define MBEDTLS_PK_PARSE_EC_EXTENDED
+#define MBEDTLS_PK_PARSE_EC_COMPRESSED
```

Comment:
Out of scope (naming): I wonder if we should keep those in the `MBEDTLS_PK` namespace - will they affect the new extended `psa_import()` API as well? If so, shouldn't they have a more neutral name?